File: BasicDemo/desktop/src/com/squidpony/basic/demo/desktop/DesktopLauncher.java
package com.squidpony.basic.demo.desktop;

import com.badlogic.gdx.Files;
import com.badlogic.gdx.backends.lwjgl.LwjglApplication;
import com.badlogic.gdx.backends.lwjgl.LwjglApplicationConfiguration;
import com.squidpony.basic.demo.BasicGame;

public class DesktopLauncher {
	public static void main (String[] args) {
    	LwjglApplicationConfiguration config = new LwjglApplicationConfiguration();
   		config.width = 80 * 12;
    	config.height = 25 * 24;
    	config.addIcon("Tentacle-16.png", Files.FileType.Classpath);
    	config.addIcon("Tentacle-32.png", Files.FileType.Classpath);
    	config.addIcon("Tentacle-128.png", Files.FileType.Classpath);
    	new LwjglApplication(new BasicGame(), config);
    }
}

File: ScalaDemo/desktop/src/main/scala/com/squidpony/basic/scala/demo/desktop/DesktopLauncher.scala
package com.squidpony.basic.scala.demo.desktop

import com.badlogic.gdx.Files.FileType
import com.badlogic.gdx.backends.lwjgl.LwjglApplication
import com.badlogic.gdx.backends.lwjgl.LwjglApplicationConfiguration
import com.squidpony.basic.scala.demo.MainApplication

/** Launches the desktop (LWJGL) application. */
object DesktopLauncher {
  def main(args: Array[String]): Unit = createApplication

  private def createApplication = new LwjglApplication(new MainApplication, getDefaultConfiguration)

  private def getDefaultConfiguration = {
    val config = new LwjglApplicationConfiguration
    config.title = "BasicScalaDemo"
    config.allowSoftwareMode = true
    config.width = MainApplication.gridWidth * MainApplication.cellWidth
    config.height = (MainApplication.gridHeight + MainApplication.bonusHeight) * MainApplication.cellHeight
    for (size <- Array[Int](128, 64, 32, 16)) {
      config.addIcon("Tentacle-" + size + ".png", FileType.Internal)
    }
    config
  }
}